### ISSUE TITLE
Error: Uninitialized string offset: -1

### ISSUE THREAD

**Issue by darkspive** [PocketMine/PocketMine-MP#3686]
```
PocketMine-MP Crash Dump Sun Nov 8 06:01:24 UTC 2015

Error: Uninitialized string offset: -1
File: /src/pocketmine/level/format/mcregion/Chunk__32bit
Line: 121
Type: notice

Code:
[112] 		unset($this->nbt->Data);
[113] 		unset($this->nbt->SkyLight);
[114] 		unset($this->nbt->BlockLight);
[115] 		unset($this->nbt->BiomeColors);
[116] 		unset($this->nbt->HeightMap);
[117] 		unset($this->nbt->Biomes);
[118] 	}
[119] 
[120] 	public function getBlockId($x, $y, $z){
[121] 		return \ord($this->blocks{($x << 11) | ($z << 7) | $y});
[122] 	}
[123] 
[124] 	public function setBlockId($x, $y, $z, $id){
[125] 		$this->blocks{($x << 11) | ($z << 7) | $y} = \chr($id);
[126] 		$this->hasChanged = \true;
[127] 	}
[128] 
[129] 	public function getBlockData($x, $y, $z){
[130] 		$m = \ord($this->data{($x << 10) | ($z << 6) | ($y >> 1)});
[131] 		if(($y & 1) === 0){

Backtrace:
#0 /src/pocketmine/level/Level__32bit(2671): pocketmine\level\format\mcregion\Chunk->getBlockId(integer 15, integer -1, integer 12)
#1 /src/pocketmine/Player__32bit(739): pocketmine\level\Level->getSafeSpawn(pocketmine\Player Player(1))
#2 /src/pocketmine/Player__32bit(719): pocketmine\Player->doFirstSpawn(boolean)
#3 /src/pocketmine/Player__32bit(1560): pocketmine\Player->sendNextChunk(boolean)
#4 /src/pocketmine/Server__32bit(2485): pocketmine\Player->checkNetwork(boolean)
#5 /src/pocketmine/Server__32bit(2222): pocketmine\Server->tick(boolean)
#6 /src/pocketmine/Server__32bit(2100): pocketmine\Server->tickProcessor(boolean)
#7 /src/pocketmine/Server__32bit(1701): pocketmine\Server->start(boolean)
#8 /src/pocketmine/PocketMine(464): pocketmine\Server->__construct(pocketmine\CompatibleClassLoader object, pocketmine\utils\MainLogger object, string phar:///storage/sdcard0/PocketMine/PocketMine-MP.phar/, string /storage/sdcard0/PocketMine/, string /storage/sdcard0/PocketMine/plugins/)
#9 (1): require(string phar:///storage/sdcard0/PocketMine/PocketMine-MP.phar/src/pocketmine/PocketMine.php)

PocketMine-MP version: 1.6dev #48 [Protocol 34; API 1.13.0]
Git commit: 0000000000000000000000000000000000000000
uname -a: Linux localhost 3.4.67 #2 SMP PREEMPT Fri Sep 26 14:22:08 HKT 2014 armv7l
PHP Version: 5.6.2
Zend version: 2.6.0
OS : Linux, android

Loaded plugins:
DevTools 1.10.0 by PocketMine Team for API(s) 1.3.1
mcg76_SimpleWorlds 1.0.1 by minecraftgenius76 for API(s) 1.0.0
SoldierGame 1.0-alpha by ifteam for API(s) 1.4.1, 1.12.0, 1.13.0, 2.0.0
essentialsTP 1.1.0 by  for API(s) 1.12.0
PocketMoney 4.0.1 by MinecrafterJPN, MinecrafterJPN for API(s) 1.10.0
SignShop 1.1.3 by xionbig, xionbig for API(s) 1.11.0
Particler 0.3 by Dark for API(s) 1.3.1

----------------------REPORT THE DATA BELOW THIS LINE-----------------------

===BEGIN CRASH DUMP===
eNrtWm1z27aW/itYObPX3hFlUZIlW73ObK6T3OZeJ9HGzuRDdScDkaCEmiJYgrSsdPPf9zkHIEU5
dtJNO7P7oZkmJYGD8/4K6tdOqdeqMw1Ho/HZeDA6HXU7qbTli6IwRWf6a6fc5tjuZKbUkep0O2tl
rVzS0vtMZ7rUMtWfVCxsWehsKUySWFVORRACNqnS9KVOCThfyWJ6fHxsS1Pg+LGNI1nE/eOZiW5U
+VpnqvUYvJ716MCxLaLjnJfXBJGqW5UeJ6ZYy/J4HRVqqU12fLGqspuPH4eDhS5xLCfCjuj//jzO
poCEPgZht1MWEjJPf+oc9MXDuC7pX3/2cDCehEdTsQOazxlqPnck5/Oa5nzORIOnS1X+LQX8q/hQ
Z6VaqkKEJ11RPwfh7jkcHIG7g/ALTmap3KqiZmIyPHuYB2aVKV7JRF3lcpMdtsEcGuH+dxgeMbXB
t6iF96m5/eBpbF7qwpaOzsKYVMmMUQ6/gTI8GfcfwWlVFr9RdyUrbw/p6AukV6q43SGFY588gjRa
qejmjSo3ptjHefItnPhzD6cDCJ4iVvZxjb+FK+z3v4JrVpgIcWeKPaSTbyANJ/3wEaS2lEW5h+z0
S7M08Xg4Go8eQfTxY2QyxH4VlXvedGHWuSz1IlUXSCf20sgYzmUWP6uo7O5hqkqd2vn8tdTZpVku
W1A+pXxf6miOf+3cbwLK02qpM3vMajoTh6TTQv1S6UId/i4WH9U3ZbGjzr8+dzvqD0nDf0Q2BDNO
EZ1pIlOrup3IxIpYC8MBsM/LeVllIHr4pFxpGzzNFmXw9Lks5dEPwBOGw0eBrm62l3q5Kj3g6FFA
TpVt0JPHQbVZqwuTmsJ62PGjsD8qQvla5h5y8nWsNcJTBvvMz2d4podBnxfzapHqSCRVFpVQqmil
+Sd3XfFki7+fjn7lE6GnVqiyKjIxn5siroku6JD9FYfEX/8qwvBI/Lc4fPKJXib0/GT72TEzGOyY
GQxrZkYPMmMfYgZ/dew5qrX625kQ5+A7WhWHhMQxNN5HspL2YiWzJVyUYJEwlIObtBg/rRk/+6oW
yam+0OOwXxNcM4WWFmPAN+z3W+yP3fNWPH0qwiOvymFtEJ0c0t6/Y0ucn5+LPgh9/rMp+LMp+LMp
+LMp+D9rCrqdpcpUIVMqvdC2ReZAwgp741jdgplFpdO4Mx2dol4XpjSRAeQQU53MNcOFw14fcEuU
92mn/xv/4EAhb1K9oDO9CWOoMkmTY+dSZ9WdQMKS6crYUgx7o954IpAmrl7PxOzdixevZ9fiZaHF
lcrFYCzC0XQwmPZPxY//vBaDfjgSsljfTlLgpPFt2jnpjXsDvH1CaON1gNe+2/xobE0RC/wis7gw
Ou40DYolzTxXt9dwZH72fDZL3T29hX1GLqtyhWaBEvtO5+JayTXpnJX3E6CHvZDeVSbhxWCOClm3
E6scrNJh7FmTlM/bC2tJXVNDfz5vcZIiBrB3df3s3fX7GRY2amF1SfyuyjK3cCBYalUtepFZt12n
wQG519FyMv54pdd5qj6YIo3bYj+wua+Afi/cl58cLypkUsLTdGUn4z0NkLq+UwPMigtuJIsWR/tv
M5jxVbZTzuzt1fWHt+8un7fVk1VpCtGvTBprVfydZW1kbq/eEzaQKeJsX2CdlPftPGKlwDsG7B1N
2Ax+j/yODoTdsbf38lV/8AIj46uM2n17PWtJvLd8z8Hv+XdbTBbvO6Vpk5zP7zHwG0znndlkatsS
pL3almP0pZ++rv1UFf+YvcHevYU9QX+H2VosoUPY4+9rYj4UwPscHreRkTPrZXa1Mnnbk+ul+zYd
7uviDjsLvcRi/bQnffj90tccUIw2vHxTbojtGellqrRoB3qmWDZls0FFfoCuQ0epKtpe0Ky15O73
hnsyP5fFzR+VnPOaICzcov2NcKSaIwuwDHvWBLgH6qH25gp4FJWog1nzJi5MluiloJl8Xsyzg6sq
E2/MrTgV/fG0H07DsXh/fUFl8YT2HbqA1HLeOM9UzF4I12y1YHJTlOfhWTgc0uISJ9aY0M/79LaW
d0HOva09H/CKpR48oBZB8XR1HoS0vFlBulTb8twkCS04ZQa/VKrYnmNI2S0VaPJqqFjZqNA5I3KM
iTWaO1FZ6oT22hxmx5Tx+QeVIiSU+E/HmCiNoHlNOHH+rSUZXX6cZzQupXwaghXbINVrDYHRnbym
Vbq1Dqocw56quZJZZjA8Ki96IKOVxsCzRqKyXpZbrTZBDHkl4ICMj6Wp2QRJSrcCNSqnLplp8FCf
dWtrs6gX0L/FkSka+vlt7ndinSQ6qtJye85qdj0cGsMA03gJHdlzloHHMTb2hkrhbs0qFbdAWCMv
L59d0xIZopejocaZ+Pw/+A+kqEoTWHmrPAts2aBtWgyg0E3jKH6VGq2m/+xt1yk5sKBOXETsuxUY
p3GcXFgAxz3zZgfieqWsErVk6O4ULKsE0pzFgyxFRJYpxUJhkI3SKlYxHsQXoUO4rshFTEII7uO0
GFm7wmCnsIRSLAolb8TWVIXHJXQiIJtONFOAbQp4e7olxG/UpkF3HKtEwjpWbEz2l1LIPFeyEKRC
jOganW26BfvOP1nuzUplosqXhYyBoDfPyFkdsuk8EwfXq0JB21gCF6nMlhVGAEFXZqwy79hW4507
Z/2JlUpHL2j2dFLsHwQDdUlRWW+jb3SuYi05sdLb8SWM+9EkH19dvf04Hp4Fg490DmpsEE3FHOlx
Oe8QoQ8rRbqjwKMhWkBKP/rATIXMbIpYit02a5M5BR94UiWbFDOHjpRYoQlP2chrIHZ+tSPJ94VY
t6uqjA2FjLu8JGZ8sohSY1Xs2HpG4SfIUXmKchVD3Gop/ouSEEA4GQV+ZyoozdNBFJSNkOSlFiOs
8FScqaRHhIwEJ0RJgCOwdM9mrwCJmhKTnnYbAdhagKkd+hec9mo8kFhQrAu4KzyCWF1shfcjgPsk
2ezu1HDwQUPRJoNLyTjG1Gid5xGbNNeDHWmrgtOUgL8tFOkDSsd4KVMyx1LfAvZWppUSh54kKsYR
kDt6SM6KykEAa9L0PMUuUX5TrRc0Syfimd1m0bW0N4KuOBBAPdp/DyOwf9ZCEtSqMEijcAlA267g
xCR8AoPLdqGGNbRmqUizWlAjaW1dZYgcAmHcV+wykAfBVZg7JFIUmi2Jw66eNayRVh07rxKKUAKh
SOwKXf4FmisLPhUrqlzshffOikNIADuLAWmEZQi8lFPGRNHqigjH6t9Ts4BiqTUQbllwbaGctFZL
udiWYEhcOVb6TFxbMq+LdMi00cwYK1tQ9fA1yi9Z1kui4S5IU8jgGaNKkCNqis78KyqZ5LRLAzkM
hQbpDHSWzKQrelPRn3Oq4KxcItXI+P85+9RmfYV5qpy/i3lbmpzp+1TV8OOxoa2oCqqRyDKeG6JZ
sxT2ByPH1UwV2nA5oli0RJAvA2vumngLQQqpJiYfY4gA8aBcpHEaNxiHMkQo9EbZ4TG1OnWiOqec
4SnWHdicCy7hQPTtItmlI6LwllIIVTgG730JTGL7ZhgqzFVEhdCyWohXp2N3thEAmlS53Yk57Hs5
7dGDDHmph07qpSwWsH8QmTR1fSVF2INaZXciVvwZ4c8Yah2ziktumwu0IxV84ahb54R7zuC9weXV
JKlzw3rnZT3x1tU7foNHrEyVxg0jGhlkk4kd6+S3ImfOIeAYjSbLKA5eciR8kRvZV9zpRqikMOs6
xdJZvx+0s1LLphDCm+6+KzzgP61z1N9HdAFunb4v2espIfIicUajDEkjXDPc5fAs0IBYBxRQD0C9
zA4XoarN7APlbOz5fG7csfvmoyJAnZI71sbgqHiwP0BmLkRBhKZeOam9L9SEaKOp3l9FNc8yd9Xf
tG5wjdRVwoUsMTVQIyKpy7X07UX4rMQB6BpTv+uS0UI1NZHTzUE7fdU7FPjFtiTcXRGELWem2sdk
g7JmBWnlZOyyyq7W8ihAhS526c4b0J3FouepJ37EJAP1nqMXhuNezN53cRYcLJB/NjouVy5rc3w3
6ANGPxUT3xjsWgbLimmB9sSzGPG0kmlyjCbfdTGxgp95QiDp/M9k7YBsqnMLV6tPenFHwUdtkEy7
LrCJ8AedoYuEWHAbxa5Ntf79LJvVncKuZ6cmiA5tkOvn1LBneeBf2z0ZDbCLajn1TcdTQZ/YSl9Z
qKvk7bqjtJzHVqruNOfeFCgjuz7R0ucDWVZwmONl5FSLvBbbFtF6tNubG1TdRAr3gdBNTAzq3YvE
ZYNHUGLJHuaPBO7IVDTfFJ1AfBFC7MLl4KSclOQefmROmsjKFnmex7/ghmOUTXfw5u31i6ngrIv/
ogrDVYbhihHxdS6ZjtnzlBv2GhM/27NWxF+Ia1nZi7gkgRPympI8Z9D3JYRSmatM5EY07NK6L0c+
6veX6xzmeuErClcScCGtahE7DBEoD1DpigE2woc2VBn1jjhorWpzwe5wTZhptPL3L0IhMTl6VPqd
JXmI3U23lAh4CqLuidKBTDdyax1uj6flR3v5m93o2dpUWbsAWN+z1UwQ+4SNRga6YcHjVIwePyoL
rMaOd4djv+qI8Gzw4Gn2rky5OXKhHLaFSigT8QUKZSwyQ4PVXau0Up/LfFfyFkWX+rv6ty6Rybf+
asDXoqah4USRSFuyVlkvPo0lVcpgxsJVAwpRHblSwiNYQfM4HANoa1X9bGgScrFh5Zp8Zc25kgpM
I/89Y5BCa2O8lndC3tdL7r7eqnjnD/u2YCd9J2Nd2dYxbwcpWpdm7qj3O4JHt0I5ie6v/I3YjkGM
2ko6InwZtKuCjvHdYFfz3vuSeZ//NlLzjI55vFLevv48hz2vBxbGmorT78KWm7xKPTb/TKXpHuJ5
xjEZQHvEtLulC9iPLGmSVLmmL9J05eJXQ58e6IZsKlxzV5uUvn9XOZod1/E5h+Tk4RoNjwtYJnTK
EySkJ/S6kUSoWeQ1uV5oeL4D8ZdzbkafumRE1wmFtCvhVl2N9U7ivNd38k1yI78O6IJnf6UpJ3ur
+SrXWWJ2bkCfaqeOZK9134dGiPnLTLZdU5vPEWKhX9swSnOL3xa7bWaYfmoj5HKJ6uMvB5prF+oR
6bczNw9L4/ghdPYhfkhfzpXZxPdPwyfc7tQZoMgCX5x3HSCvmrx1YXRBd4Z8iaVSk1O16uIdEtAN
XVm3YuhMEnTKCspGfcqozhOQL63uhlu2kQj/xaLVQ1RoOG3pClyd8Dw2spDbrvHbe0rZbDYPqSTV
KDfOKi/uJNVb90INC/B7JqY1N7wFZ6dJeSp+oscARabLo/O/CKPLgnUb0r5tZSHqecENbfUFNk9j
D13e8nDLVcxyx0Bir9FIaDDqS3yLdabKDPCToPtuCl0uSTtqU0GX3tPBD5Puydld2B3eDbuDH8If
YlRil7UOy0LhBRnm/LQvlvBv695GJ0d8ua3uSpWRQvgj+QXqUOsXB+RCrVeUg2Xze8c8IlD/Yn9J
damG7ucQgfvpxSf3+wj3xXgRoa9Z7X4sKVNEjix2C+4XnfVbVaTNS6LTkr6GAXXoPt4mZd7sLte7
5xWi133WBtDP1n8cHPCH0vXC3eU2wOgJt3m5k6CUxFDcEvdqdslU6TmP0Mvtzm4hMP8Ywz+JE/oe
G5LkIqCvVYP+SX+IxyevYLdkOBkqeRLHJyfD08k4HPXH8rQ/GMjR4lQOJ6NReLoYjPpKPAGl2YrU
wl/WmbCx+m6ndD+se4CQBH2nEj9pYpXpnaqTJJLoVEcg208mk2hyejo+OQHp4dlgqIb9xdnkLI6Z
nl2tzU6FlgPL7kuqnfIBXJoblaHOFM3KByVvkBGcononWNlK/khCX2UJ4JPO3dugN+p8/vw/PmsJ
YQ==
===END CRASH DUMP===
```

**Comment by darkspive** [comment]
I tried reinstall map.
Does not works.
Downloaded from jenkins

**Comment by iamstewpeed** [comment]
Hey There AlwaysSpawn Plugin will Fix that Crash Dump

**Comment by TahaTheHacker** [comment]
@iamstewpeed this is not a forum.

**Comment by Charelene** [comment]
@TahaTheHacker This is dead-project-chat, right ![image](https://cloud.githubusercontent.com/assets/12827721/11326728/b2424560-9179-11e5-914b-1d7c883e9c65.png)

**Comment by da123rrell** [comment]
but seriously when will it be fixed?

**Comment by 3do2** [comment]
I^have the same issue, 10 times per day, every days since a while :/

**Comment by YourturnGaming** [comment]
Had a customer with this issue, what I did to fix it was to delete the player data, this reset them. They were later to go into that chunk and was able to move around freely. Don't know how this error gets produce in the first place, any knowledge would be helpful

-Austin | YourturnGaming

**Comment by 3do2** [comment]
@YourturnGaming Add AlwaysSpawn Plugin to your server. That will reset player spawn position at respawn. And the bug won't happened again.

**Comment by YourturnGaming** [comment]
Cheers for the quick reply.

**Comment by RocoWayne** [comment]
yes! alwayspawn is the solution!

**Comment by da123rrell** [comment]
not if your running a survival server and the players like to roam around the world. they will get pissed off if the server restarted and they lost their house.

**Comment by Intyre** [comment]
If anyone can provide steps to reproduce the error it would get fixed faster. Using a plugin won't magically fix the real issue.

**Comment by robske110** [comment]
Ok @Intrye 
This issue is caused by some wrong calaculation with the y pos.
You can easily reproduce it:
Start an new fresh server with no plugins (creative mode)
Fly 10 chunks away from spawn
Dig down to the bedrock, stand directly on it (y pos should be 2,1,0)
Log out
Log in >crash

**Comment by PEMapModder** [comment]
Are you sure that the exactly same error occurs in the latest PocketMine version?

**Comment by robske110** [comment]
Sadly my main servers aren't on pocketmine anymore (Due to: Buggy, NewBuilds have to be self-compiled...)
However some of my test servers for plugins are on pocketmine, I will test it today and post here if i could reproduce it! (Will use the phar from the 0.13.2 commit)

**Comment by PEMapModder** [comment]
Fixed in #3956
